handlers/dimensions.go

```diff
@@ -511,79 +497,74 @@ func (f *Filter) addAll(w http.ResponseWriter, req *http.Request, redirectURL st
 }
 
 // AddList adds a list of values
-func (f *Filter) AddList(w http.ResponseWriter, req *http.Request) {
-
-	vars := mux.Vars(req)
-	name := vars["name"]
-	filterID := vars["filterID"]
-	ctx := req.Context()
-	collectionID := getCollectionIDFromContext(ctx)
-	userAccessToken, err := headers.GetUserAuthToken(req)
-	if err != nil {
-		if headers.IsNotErrNotFound(err) {
-			log.Event(ctx, "error getting access token header", log.WARN, log.Error(err))
+func (f *Filter) AddList() http.HandlerFunc {
+	return dphandlers.ControllerHandler(func(w http.ResponseWriter, req *http.Request, lang, collectionID, userAccessToken string) {
+		vars := mux.Vars(req)
+		name := vars["name"]
+		filterID := vars["filterID"]
+		ctx := req.Context()
+
+		if err := req.ParseForm(); err != nil {
+			log.Event(ctx, "failed to parse form", log.ERROR, log.Error(err), log.Data{"filter_id": filterID, "dimension": name})
+			setStatusCode(req, w, err)
+			return
 		}
-	}
 
-	if err := req.ParseForm(); err != nil {
-		log.Event(ctx, "failed to parse form", log.ERROR, log.Error(err), log.Data{"filter_id": filterID, "dimension": name})
-		setStatusCode(req, w, err)
-		return
-	}
-
-	redirectURL := fmt.Sprintf("/filters/%s/dimensions", filterID)
-
-	if len(req.Form["add-all"]) > 0 {
-		redirectURL = fmt.Sprintf("/filters/%s/dimensions/%s", filterID, name)
-		f.addAll(w, req, redirectURL)
-		return
-	}
-
-	if len(req.Form["remove-all"]) > 0 {
-		redirectURL = fmt.Sprintf("/filters/%s/dimensions/%s/remove-all", filterID, name)
-		http.Redirect(w, req, redirectURL, 302)
-		return
-	}
+		redirectURL := fmt.Sprintf("/filters/%s/dimensions", filterID)
 
-	var wg sync.WaitGroup
-	wg.Add(1)
+		if len(req.Form["add-all"]) > 0 {
+			redirectURL = fmt.Sprintf("/filters/%s/dimensions/%s", filterID, name)
+			f.addAll(w, req, redirectURL)
+			return
+		}
 
-	// concurrently remove any fields that have been deselected
-	go func() {
-		opts, err := f.FilterClient.GetDimensionOptions(ctx, userAccessToken, "", collectionID, filterID, name)
-		if err != nil {
-			log.Event(ctx, "failed to get options from filter client", log.ERROR, log.Error(err), log.Data{"filter_id": filterID, "dimension": name})
-			setStatusCode(req, w, err)
+		if len(req.Form["remove-all"]) > 0 {
+			redirectURL = fmt.Sprintf("/filters/%s/dimensions/%s/remove-all", filterID, name)
+			http.Redirect(w, req, redirectURL, 302)
 			return
 		}
 
-		for _, opt := range opts {
-			if _, ok := req.Form[opt.Option]; !ok {
-				if err := f.FilterClient.RemoveDimensionValue(ctx, userAccessToken, "", collectionID, filterID, name, opt.Option); err != nil {
-					log.Event(ctx, "failed to remove dimension values", log.WARN, log.Error(err))
+		var wg sync.WaitGroup
```

Comment:
So I would remove the wg variable from this function

Reply:
I have changed the current annotation:

> // concurrently remove any fields that have been deselected
To
> // TODO concurrently remove any fields that have been deselected

So that the information you have provided isn't lost I will raise this as an issue against the repository.

Reply:
In the interim is it safe to remove all of the Wait group parts as there is nothing going on in parallel here?
```
var wg sync.WaitGroup
wg.Add(1)
wg.Done()
wg.Wait()
```

Reply:
Yes that is correct - remove those 4 lines 👍🏼